waspc/cli/src/Wasp/Cli/Parser/Util.hs

```diff
@@ -0,0 +1,22 @@
+module Wasp.Cli.Parser.Util (CommandType (..), mkCommand, mkWrapperCommand) where
+
+import Options.Applicative (Parser)
+import qualified Options.Applicative as O
+
+data CommandType = CTNormal | CTNoIntersperse | CTForwardOptions
```

Comment:
Thank you! :+1:  Hmm, that's true with the prefix and modifiers. This actually prompts me re-think this a bit because optparse-applicative have [more](https://hackage.haskell.org/package/optparse-applicative-0.18.0.0/docs/Options-Applicative.html#t:InfoMod) info modifiers provided than by my new `CommandType`. 

This current code forces the code user to update the type and the `withModifier` internal function if they want to add different modifier in the future. So, I decided to pass a list of modifiers instead.

What I'm thinking I wanna go for right now is:

```haskell

mkWrapperCommand :: String -> [InfoMod a] -> Parser a -> String -> O.Mod O.CommandFields a
mkWrapperCommand name infoModifiers callCommand description =
  O.command
    name
    ( O.info
        (O.helper <*> callCommand)
        (O.progDesc description <> mconcat infoModifiers)
    )
```

This way, they can add more than one modifiers if they choose to do so and doesn't need to update multiple things at once.

> Another question, are we sure we need both and not just forward options?

You're correct! I think we only need `forwardOptions`. My reasoning for `deploy` command to use `noIntersperse` was to reject something like `wasp deploy --flag` early on because I don't see anything like this being used in the [docs](https://wasp-lang.dev/docs/deploying). 

I also thought if `deploy` provided with arguments, it need to start with at least, one tool name (`wasp deploy deployToolName --blah something etc -v`).

Reply:
Aha, well
> This is the same behaviour one sees after the "--" pseudo-argument.
this sounds good.

Ok, I will let you investigate this a bit more, but it would make sense to me if we can have this special behaviour only for specific commands, for which we know that they forward their arguments. If we can't fine, that is survivable, but if we can that sounds cleaner.

Reply:
@martinsos sorry. :sweat_smile:  I believe I've read your question a bit wrong. By this:

> Turn them on only for wasp deploy though and disable them for the rest of the CLI

I think it already works this way for deploy. Currently, only `wasp test client` and `wasp deploy` use `forwardOptions` to treat unknown arguments as positionals (we could make this even looser with `allPositional`, which I think makes more sense if we really want `node` to handle ***all*** arguments for these two commands).

The rest of the CLI commands (such as new, etc) won't be affected by this. 

btw, I think we should use `allPositional` for deploy.

Examples of `forwardOptions` vs `allPositional` for `deploy`:

## forwardOptions
This will still exit and print our own optparse-applicative help message if we pass help flag.
```sh
$ wasp-cli deploy --help

Usage: wasp-cli deploy [DEPLOY_ARGUMENTS]

  Deploys your Wasp app to cloud hosting providers.

Available options:
  -h,--help                Show this help text
  DEPLOY_ARGUMENTS         Currently only supports fly.io. See
                           https://wasp-lang.dev/docs/deploying.

```

## allPositional
This will pass ***all*** arguments after `wasp deploy` to node. This is also the exact same behavior before this PR.
```sh
$ wasp-cli deploy --help

Usage: wasp deploy [options] [command]

CLI for deploying Wasp apps to various clouds

Options:
  -h, --help      display help for command

Commands:
  fly             Create and deploy Wasp apps on Fly.io
  help [command]  display help for command

```

Reply:
Ok sounds great that it is already limited to just those two commands, and very nice analysis here @modotte -> all sounds reasonable, allPositional seems like obvious winner here then! You understand this part better than me at this point so whatever you find better, I am ok with.